sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +764,84 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
```

Comment:
From design standpoint, I am thinking that our poll result type should be SearchStatus instead of PhoneNumberSearch.. so then this declaration should change to -- 
public PollerFlux<SearchStatus, PhoneNumberSearch> beginCreateSearch

Reply:
The only thing about this is that we we wouldn't have a searchId to look for in the polling operation because we would be returning a SearchStatus, that doesn't have a searchId as a property

Reply:
And also, our polling operation would have to return a SearchStatus and that wouldn't help in the fetchResultOperation, that just takes the final response from the polling operation and return it.